hmy/staking.go

```diff
@@ -199,6 +199,33 @@ func (hmy *Harmony) GetAllValidatorAddresses() []common.Address {
 	return hmy.BlockChain.ValidatorCandidates()
 }
 
+var (
+	epochBlocksMap = map[uint64]staking.EpochSigningEntry{}
+)
+
+func (hmy *Harmony) getEpochSigning(epoch *big.Int, addr common.Address) (staking.EpochSigningEntry, error) {
+	entry := staking.EpochSigningEntry{}
+	if val, ok := epochBlocksMap[epoch.Uint64()]; ok {
+		return val, nil
+	}
+
+	snapshot, err := hmy.BlockChain.ReadValidatorSnapshotAtEpoch(epoch, addr)
+	if err != nil {
+		return entry, err
+	}
+
+	// the signing information is for the previous epoch
+	entry.Epoch = big.NewInt(0).Sub(epoch, common.Big1)
+	entry.Blocks = snapshot.Validator.Counters
```

Comment:
Better do the diff between counters and show the signed/to-sign per epoch, instead of showing those throughout the history.

Reply:
I don't want to handle the precision at the backend. Passing the raw values will help dashboard change its logic whenever required.

Reply:
I am not saying calculating the uptime here. I am saying to return the to-sign/signed for the period of the epoch, instead of returning the full stats up until that epoch.

Reply:
got it.